enhancements/rhcos/static-networking-enhancements.md

```diff
@@ -0,0 +1,104 @@
+  - "@cgwalters"
+  - "@crawford"
+  - "@imcleod"
+  - "@runcom"
+creation-date: 2020-04-22
+last-updated: 2020-04-22
+status: provisional
+see-also: https://github.com/openshift/enhancements/pull/210
+replaces:
+superseded-by:
+---
+
+# Improved Static Networking Configuration
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+This enhancement describes improvements to how static networking is configured on Fedora CoreOS (FCOS) and RHEL CoreOS (RHCOS).  It covers improving the configuration of static networking for bare metal and VMWare hosts.
+
+## Motivation
+Many customer environments require static networking and do not allow any type of DHCP server. This is most prominent in VMware environments where they struggle with the OVA images and are resisting the bare metal installer (via ISO) approach. This is also a challenge in bare metal environments due to an unfriendly and time sensitive interactive flow (i.e. catching the `grub` prompt). Due to the intersections of Ignition, Platform ID, and active networking in the `initrd`, many of the current RHCOS images append `ip=dhcp,dhcp6` and catching the bootloader on the console is not a pleasant experience. (Even if a static `ip=::::` is passed it will fail as the platform ID will insert `ip=dhcp` and conflict)
+
+This is basic networking configuration and users assume basic configurations like this are possible. As RHCOS requires a functional networking config early in the boot process, this eliminates many of the existing mechanisms customers use to handle the assigning of static networking configs. Most of the OpenShift personas we target are not aware of `dracut` arguments and it’s not a great experience to expect them to understand very low-level details of how the OS works.
+
+### Goals
+  - Provide a guided configuration of networking and Ignition source information
+  - Provide tooling to create customized ISOs that include unique Ignition configurations
+  - Provide support for a VMWare "backchannel" to provide network configuration and Ignition source information
+
+### Non-Goals
+  - Providing a fully automated solution for providing static network configuration to a fleet of machines
+
+## Proposal
+For the bare metal use case, we are proposing the use of the new [live ISO](https://github.com/openshift/enhancements/pull/210) to provide an interactive environment where users can configure their networking parameters and have them persist to the real root.  In this environment, users will be able to invoke a TUI that will allow them to configure networking and Ignition source.  This information will be used as part of the install of the host and will persist into the real root.
```

Comment:
> to provide an interactive environment where users can configure their networking parameters and have them persist to the real root
how about:

"to provide an interactive install environment where users can configure their networking parameters and have them persisted into the installed system".

> In this environment, users will be able to invoke a TUI that will allow them to configure networking and Ignition source.

Note that phase 1 of this doesn't involve the "enhanced TUI", just `nmtui`. Assuming that is OK.

> This information will be used as part of the install of the host and will persist into the real root.

We can probably just delete this line if you accept the first suggestion. If we don't delete this line I think we should reconsider terms like `real root`, since it can be too specific for people without context. "Installed system" is more effective.

Reply:
> Note that phase 1 of this doesn't involve the "enhanced TUI", just nmtui. Assuming that is OK.

IMHO that should be fine.

Reply:
Got these changes locally...updating soon.